models/Asset.php

```diff
@@ -1562,28 +1562,15 @@ public function getMetadata($name = null, $language = null, $strictMatch = false
             $data = null;
             foreach ($this->metadata as $md) {
                 if ($md['name'] == $name) {
-                    if ($language == $md['language']) {
-                        if ($raw) {
-                            return $md;
-                        }
-
-                        return $convert($md);
-                    }
-                    if (empty($md['language']) && !$strictMatch) {
-                        if ($raw) {
-                            return $md;
-                        }
+                    if ($language == $md['language'] || (empty($md['language']) && !$strictMatch)) {
                         $data = $md;
+                        break;
```

Comment:
This cannot break in the case of `empty($md['language']) && !$strictMatch`, see:
https://github.com/pimcore/pimcore/pull/12800#issuecomment-1232750617